### DISCUSSION TITLE
health check of the app infrastructure

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#1109] [Ideas]
Adding a  basic health probe configuration that reports the app's availability to process requests (liveness) is sufficient to discover the status of the app.  Have this simple endpoint for monitoring services such as [Uptime](https://uptime.com/)

obviosity checks could be extend to monitor the database(s).

This would be similar to the KeepAlive service found in DNN

[Health checks in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-5.0)

**Comment by leigh-pointer** [comment]
I should also add I am happy to implement this.

**Comment by hishamco** [comment]
I'm agree to add Health check, but we should schedule it in the roadmap

/cc @sbwalker

**Comment by sbwalker** [comment]
@leigh-pointer I assume you are referring to a public Web API endpoint which simply reports if the Server is operational? Can you describe how this would be utilized?

**Reply by leigh-pointer** [reply]
@sbwalker Yes I am reffering to the public WebAPI endpoint or endpoints. 
Health checks are usually used with an external monitoring service or container orchestrator to check the status of an app. We can configure the health check for Oqtane framework and link to any external UI monitoring (for example: https://uptimerobot.com/ , https://uptime.com/ ) or add AspNetCore.HealthChecks.UI NuGet package to the web project to create UI for monitoring. 

There are three types of health status that are returned back to the monitoring service which uses the health check.

- Degraded - Indicates that the component was in a degraded state.
- Healthy - Indicates that the component was healthy.
- Unhealthy - Indicates that the component was unhealthy, or an unhandled exception was thrown while executing the health check.

In the real world this basic health check won't be enough to know the status of the app, we need to monitor the health of dependencies of the framework like a database, storage and external API.

There are many  health check packages ; SQLServer, MongoDb. Uris. AzureStorage, AzureServiceBus. This would imply a change to the startup.cs to configure this service and maybe appsettings.config to enable or disable the service (as an option).
